[feat] Issue Template Checklist Checker

hspinks opened the pull request:
Creates a GitHub Action to automatically label and optionally comment on Issues that do not fill out the default checklists included in the GitHub Issue templates. Pulled out of WorldHealthOrganization/app#1081

advayDev1 commented:
I know this is a draft but if there is anyway not to have the dist in there, would be great.  Otherwise please make sure everything copied into the dist is properly licensed/attributed.

hspinks commented:
@advayDev1 I believe that the final built version does need to be committed to the repo, both to publish to the GitHub Marketplace, but also to be able to test the non-published versions of the code in workflows within the repo (see 10359ea). In this action, the only dependencies included in the final script are both official [GitHub Actions toolkit](https://github.com/actions/toolkit) libraries built specifically to be used in actions and distributed under the MIT license. Let me know what I'd need to do to include them in the repo.

On a separate note, looking at [the docs](https://help.github.com/en/actions/building-actions/publishing-actions-in-github-marketplace), it looks like we'll have to set this up differently to distribute using the GitHub Marketplace. It looks like each action needs its own standalone repo. Do we want to set up separate repos for every action we'd want to create? If @SamMousa is down for that, I am, but if you guys think it's simpler to not create a bunch of new WHO repos, I'm fine moving this to a personal repo.

SamMousa commented:
Regarding compiled asset in repo, we should use the docker image feature: https://help.github.com/en/actions/building-actions/creating-a-docker-container-action

That way github will build a docker image (in which we build our TS code) and then use that image for doing the action.

SamMousa commented:
What about bundling our actions in 1 action and choosing which one to use with a parameter?
It's not ideal but could be a good tradeoff? 
```
uses: who/app-actions
with:
  command: pr-check
```

Or something like that?
```

hspinks commented:
Hmm... both of these seem like a bit of unnecessary yoga to accomplish something relatively simple. Docker image would work, but seems pretty unnecessary just to avoid including 2 MIT-licensed packages built specifically by GitHub to be used in actions that make their own platform richer. If we implemented the 2nd approach of bundling all actions, then I could see how it'd be a risk if we added new dependencies later, but that approach feels unnecessary to me, too.

I'm thinking that it's probably best for me to just move these to a personal repo. I can make you two admins on that repo so I wouldn't be a bottleneck, and then in the app repo, we can lock the dependency on a specific version of the action. It just feels like we're spending a lot of time/thought on something that is orthogonal to the value of the app...
